drivers/volume/portworx/portworx.go

```diff
@@ -1050,6 +1055,33 @@ func (d *portworx) UpgradeDriver(version string) error {
 	return nil
 }
 
+// GetClusterPairingInfo return underlying storage details
+func (d *portworx) GetClusterPairingInfo() (map[string]string, error) {
+	pairInfo := make(map[string]string)
+	pxNodes, err := d.schedOps.GetRemotePXNodes(remoteKubeConfigPath)
+	if err != nil {
+		logrus.Errorf("err retrieving remote px nodes: %v", err)
+		return nil, err
+	}
+
+	clusterMgr, err := d.getClusterManagerByAddress(pxNodes[0].Addresses[0])
+	if err != nil {
+		return nil, err
+	}
+	resp, err := clusterMgr.GetPairToken(false)
+	if err != nil {
+		return nil, err
+	}
+	logrus.Infof("Response for token: %v", resp.Token)
+
+	// file up cluster pair info
+	pairInfo[clusterIP] = pxNodes[0].Addresses[0]
```

Comment:
pxNodes can be an empty array, so the length should be checked before accessing it.